ranger_base/src/ranger_messenger.cpp

```diff
@@ -491,4 +500,27 @@ double RangerROSMessenger::ConvertCentralAngleToInner(double angle) {
   phi_i *= angle >= 0 ? 1.0 : -1.0;
   return phi_i;
 }
+
+bool RangerROSMessenger::TriggerParkingService(
+    ranger_msgs::TriggerParkMode::Request& req,
+    ranger_msgs::TriggerParkMode::Response& res) {
+  // Call to trigger park mode
+  if (req.TriggerParkedMode) {
+    res.response = true;
+    res.isParked = true;
+    robot_->SetMotionCommand(0.0,
+                             0.0);  // This functions needs to be invoked before
+                                    // the parking mode can be triggered
+    robot_->SetMotionMode(MotionState::MOTION_MODE_PARKING);
+  } else {  // Call to release park mode
+    res.response = true;
+    res.isParked = false;
+    robot_->SetMotionMode(MotionState::MOTION_MODE_DUAL_ACKERMAN);
+    robot_->SetMotionCommand(
+        0.0, 0.0);  // Setting the mode to dual Ackerman doesn't return the
+                    // wheels to its original position, hence this function.
+  }
+  parking_mode_ = res.isParked;
+  return res.response;
```

Comment:
This should also always return true, i think

```suggestion
  return true;
```